### PR TITLE
Use crosstestimpl.T instead of testing.T in all PF crosstests

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Redoing https://github.com/pulumi/pulumi-terraform-bridge/pull/2583 on top of the existing PR stack in the area.
The `crosstestimpl.T` interface is compatible with both `testing.T` and `rapid.T`, allowing us to use cross-tests in rapid property-based tests.

### CODE REVIEW

File: dynamic/go.mod [hunk 1]
```diff
@@ -26,10 +26,16 @@ require (
 
 require (
 	github.com/fsnotify/fsnotify v1.7.0 // indirect
+	github.com/hashicorp/logutils v1.0.0 // indirect
```
[the remark this answers is not in the document]
Comment (t0yv0) [reply dynamic/go.mod:29]: Curious does dynamic still need a separate go.mod after the grand go.mod unification?

**Review by t0yv0 — APPROVED** [review]
Looking good at a glance.

File: dynamic/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (VenelinMartinov) [reply dynamic/go.mod:29]: Dynamic uses the pf pulcheck module, which now depends on the crosstestimpl module. I think that's why the dependency is pulled in.

I'm not sure why dynamic still needs a separate go.mod, can we get rid of it too?

[the remark this answers is not in the document]
Comment (iwahbe) [reply dynamic/go.mod:29]: We should be able to. Separate PR though.

**Pull request merged**